src/components/ModalConfirmDelete.js

```diff
@@ -0,0 +1,48 @@
+import Modal from "./Modal.js";
+
+export default class ModalConfirmDelete extends Modal {
+  constructor(modalSelector) {
+    super(modalSelector);
+    this._modalForm = this._modalElement.querySelector(".modal__form");
+    this._modalSubmitButton = this._modalElement.querySelector(
+      ".modal__button-confirm"
+    );
+  }
+
+  //Public method setDeleteConfirmation()
+  setDeleteConfirmation(handleFormSubmit) {
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  //Public method setEventListeners()
+  setEventListeners() {
+    this._modalForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleFormSubmit();
+      this._modalForm.reset();
+      this.close();
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well